### PR TITLE
Add blob size checks

### PR DESCRIPTION
**Pull request by Kern--**
Signed-off-by: Kern Walster <walster@amazon.com>

*Issue #, if available:*
N/A

*Description of changes:*
This change adds validations that a zinfo blob's claimed size matches the actual byte buffer size.

It also updates gzip zinfo tests to validate some more edge cases.

*Testing performed:*
`make check && make test && make integration`

Manually tested a pre-existing index that uses the v1 gzip zinfo format to verify that it's still rpull-able.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: compression/zinfo.c [hunk 1]
```diff
@@ -588,6 +593,18 @@ struct gzip_zinfo* blob_to_zinfo(void* buf, offset_t len)
     memcpy(&span_size, cur, 8);
     cur += 8;
 
+    claimed_size = PACKED_CHECKPOINT_SIZE * decode_int32(size) + BLOB_HEADER_SIZE;
+    if (claimed_size == len) {
+        // If we have exactly size checkpoints, then we have a current blob
```
Comment (djdongjin) [inline compression/zinfo.c:598]: I'm think if `ZINFO_VERSION_CUR` is necessary. This might only work for `ZINFO_VERSION_TWO` given we might change something else breaking this in future `ZINFO_VERSION_THREE`, and we need to change it to `ZINFO_VERSION_TWO` (and comment `current` -> `v2`)

File: compression/gzip_zinfo_test.go [hunk 2]
```diff
@@ -37,6 +37,36 @@ func TestNewGzipZinfo(t *testing.T) {
 			checkpoints: []byte{},
 			expectError: true,
 		},
+		{
+			name:        "checkpoints with less than 'header size' bytes header should return error",
+			checkpoints: []byte{00},
+			expectError: true,
+		},
+		{
+			name: "checkpoints with too few checkpoints should return error",
```
Comment (djdongjin) [inline compression/gzip_zinfo_test.go:46]: nit: I feel `checkpoints with too few checkpoints` (and `checkpoints with zero checkpoints`) reads confusing. I guess the first `checkpoints` means `zinfoBytes` passed to `NewGzipZinfo` and the second `checkpoints` means `checkpoints` within the zinfo(Bytes). We probably can change the arg name `func NewGzipZinfo(checkpoints []byte)` to `func NewGzipZinfo(zinfoBytes []byte)`.

**Review by djdongjin — DISMISSED** [review]
Left 2 comments, they have no functional impacts so not blcokers.

**Review by djdongjin — APPROVED** [review]

**Review by sbuckfelder — APPROVED** [review]
LGTM
